### ISSUE TITLE
Finishing-Upgrade (STATE_CHANGED) with healthcheck enabled services

### ISSUE THREAD

**Issue by furkanmustafa** [rancher/rancher#7786]
**Rancher Versions:**
Server: 1.3.3
healthcheck: 0.2.0
ipsec: 0.0.2
network-services: 0.0.8
scheduler: 0.3.0

**Docker Version:** 1.12.6, 1.13.0

**OS and where are the hosts located? (cloud, bare metal, etc):** Debian 8 / Arch, Both under qemu

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)** single node rancher with external db

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** cattle

**Steps to Reproduce:**
 - Make a new stack
 - Add a service
   - Enable healthcheck for the service, set some long count/time for healthy checks. like; 10 times with 6 seconds interval.
 - When up and running, start upgrading the service
 - Old container will be stopped, new container will start, will first be in "Initializing" state, until enough healthchecks are succeeded.
 - But upgrading UI will say "Upgraded" and allow "Finish Upgrade" when before container is considered healthy.
 - Press "Finish Upgrade" 

**Results:** 
 - Upgrading state will switch to "Finishing-Upgrade (STATE_CHANGED)", and will be stuck like that
 - Container state will be stuck at "Initializing" even though it becomes healthy
 - Only "Cancel Upgrade" option is available. *(edited)*

**Expected Behavior:**

Either;
 - Accept "Initializing" state as OK and finish upgrade cleanly.
 - Or wait until all new containers are determined "healthy"/running before saying "Upgraded" and allowing "Finish Upgrade" action.

**Comment by furkanmustafa** [comment]
I have tried this a couple times more, and reporting for accuracy; It did recover to "Active" state after waiting for a while (10~20 seconds).

Means; state is being broken with this steps only for a short time in my last tests.

**Comment by jzakrzeski** [comment]
I experienced this same issue upgrading a load balancer service this morning.

Component | Version
----------- | ----------
Rancher | v1.3.4
Cattle | v0.175.10
User Interface | v1.3.6
Rancher Compose | v0.12.1

Infrastructure Services:
Service | Version
-------- | ---------
healthcheck | v0.2.0
net | v0.8.1
metadata | v0.7.0
network-manager | v0.4.0
scheduler | v0.5.1

Using rancher CLI I did an upgrade from a docker compose file to bump rancher/lb-service-haproxy from 0.4.6 to 0.4.9. When I clicked "Confirm Upgrade," the state changed to Finishing-Upgrade STATE_CHANGED. Cancelling and attempting to roll-back caused rancher to continuously start new containers (using the 0.4.9 image, not the 0.4.6 that existed previously), and stop the "old" 0.4.9 containers continuously. Eventually, it stopped, and the entire service disappeared from the stack. I was then able to redeploy via the rancher CLI, and confirm the upgrade with no issue.

Trying again on a different environment, I saw the same behavior, but waiting about 30 seconds allowed it to continue successfully.

These load balancers have scheduler affinity set to global, and a small amount of custom HAProxy config to set a default backend for redirect. The internals of the container seemed fine, logging revealed that they were continually receiving SIGTERM so I don't think that it had anything to do with the load balancer service itself, appears rancher itself just lost state.
